srcpkgs/goldendict-ng/template

```diff
@@ -0,0 +1,19 @@
+# Template file for 'goldendict-ng'
+pkgname=goldendict-ng
+version=24.05.05
+revision=1
+archs="x86_64"
```

Comment:
why only x86_64?

Reply:
I don't have an option to test other archs. If there is an EASY way to do it I will be happy to do it. I wanted to start with something and to add when there will be a deamend or when I have the option.

Reply:
doesn't matter, archs= should only be used if something absolutely won't work.

Reply:
**For i686 I got:** Package 'qt6-webengine-devel-6.7.2_1' not found in repository pool.
**For aarch* I got:** ERROR Failed to execute command /void-packages/xbps-src (Exec format error).

Do I need to do anything else?

Reply:
sounds like you tried to do a native build on aarch64, you want to do cross build

i686 doesn't need to be explicitly removed because the inability to build a dependency will exclude it on that arch

Reply:
I ran the following command after running the bootstrap which is the similar to what I did for MUSL:
`./xbps-src -A aarch64 pkg goldendict-ng`

The bootstrap command ended with the same error.

What should I do different? Can you point me to the right documentation?

I am looking currentlly at this documents:
https://github.com/void-linux/void-packages/blob/master/Manual.md
https://github.com/void-linux/void-packages?tab=readme-ov-file

Reply:
Also arm not relevant. Some parts of the QT6 package are missing also there:

> CMake Error at /usr/aarch64-linux-gnu/usr/lib/cmake/Qt6/QtPublicDependencyHelpers.cmake:216 (find_package):
>   Could not find a package configuration file provided by "Qt6HostInfo" with
>   any of the following names:
> 
>     Qt6HostInfoConfig.cmake
>     qt6hostinfo-config.cmake
> 
>   Add the installation prefix of "Qt6HostInfo" to CMAKE_PREFIX_PATH or set
>   "Qt6HostInfo_DIR" to a directory containing one of the above files.  If
>   "Qt6HostInfo" provides a separate development package or SDK, be sure it
>   has been installed.
> Call Stack (most recent call first):
>   /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt6/Qt6Dependencies.cmake:14 (_qt_internal_find_host_info_package)
>   /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt6/Qt6Config.cmake:134 (include)
>   CMakeLists.txt:46 (find_package)
> 
> 
> CMake Error at CMakeLists.txt:46 (find_package):
>   Found package configuration file:
> 
>     /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt6/Qt6Config.cmake
> 
>   but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
>   FOUND.
> 
> 
> -- Configuring incomplete, errors occurred!
> => ERROR: goldendict-ng-24.05.05_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${build_wrksrc}' exited with 1
> => ERROR:   in do_configure() at common/build-style/cmake.sh:77

Reply:
Do I need to do anything else regarding the platforms testing?

Reply:
> Some parts of the QT6 package are missing also there

that means you're missing a hostmakedepend/makedepend, you can use xlocate to find the cmake file it wants

Reply:
After adding the missing package to makedepends the foloowing message is shown:

> ERROR: goldendict-ng-24.05.05_1: this package cannot be built for aarch64.

Reply:
remove the archs= line

Reply:
You are right. I forgot I re-added it. I get the same error as before after adding the qt6-base package to makedepends and also tried to add it to hostdepends.

> CMake Error at /usr/aarch64-linux-gnu/usr/lib/cmake/Qt6/QtPublicDependencyHelpers.cmake:216 (find_package):
>   Could not find a package configuration file provided by "Qt6HostInfo" with
>   any of the following names:
> 
>     Qt6HostInfoConfig.cmake
>     qt6hostinfo-config.cmake
> 
>   Add the installation prefix of "Qt6HostInfo" to CMAKE_PREFIX_PATH or set
>   "Qt6HostInfo_DIR" to a directory containing one of the above files.  If
>   "Qt6HostInfo" provides a separate development package or SDK, be sure it
>   has been installed.
> Call Stack (most recent call first):
>   /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt6/Qt6Dependencies.cmake:14 (_qt_internal_find_host_info_package)
>   /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt6/Qt6Config.cmake:134 (include)
>   CMakeLists.txt:46 (find_package)
> 
> 
> CMake Error at CMakeLists.txt:46 (find_package):
>   Found package configuration file:
> 
>     /usr/aarch64-linux-gnu/usr/lib64/cmake/Qt6/Qt6Config.cmake
> 
>   but it set Qt6_FOUND to FALSE so package "Qt6" is considered to be NOT
>   FOUND.
> 
> 
> -- Configuring incomplete, errors occurred!
> => ERROR: goldendict-ng-24.05.05_1: do_configure: 'CFLAGS="-DNDEBUG ${CFLAGS/ -pipe / }" CXXFLAGS="-DNDEBUG ${CXXFLAGS/ -pipe / }" cmake ${cmake_args} ${configure_args} ${LIBS:+-DCMAKE_C_STANDARD_LIBRARIES="$LIBS"} ${LIBS:+-DCMAKE_CXX_STANDARD_LIBRARIES="$LIBS"} ${wrksrc}/${build_wrksrc}' exited with 1
> => ERROR:   in do_configure() at common/build-style/cmake.sh:77

Reply:
I think `qt6-network(-devel)` is needed? The qt6-base package seems splitted in voidlinux.

Reply:
Not clear why I would want to add that package, but after adding it to the makedepends I get the following error:

> => ERROR: goldendict-ng-24.05.05_1: target dependency 'qt6-network-devel' does not exist!

I add add it also to the hostdepends I get the same error as before. If I add only to host depends I also get the same error as before.

Reply:
I don't have void installed now, but the `makedepends` of others software is just `qt6-netowrk`? I don't really know.
https://github.com/search?q=repo%3Avoid-linux%2Fvoid-packages%20qt6-network&type=code

Reply:
> Not clear why I would want to add that package,

To fix this error

>  Could not find a package configuration file provided by "Qt6HostInfo" with

I don't know the exact package name, something qt network.

Reply:
```
$ xlocate Qt6HostInfo
qt6-base-6.7.2_1  /usr/lib/cmake/Qt6HostInfo/Qt6HostInfoConfig.cmake
```

Reply:
Did that already here:
https://github.com/void-linux/void-packages/pull/51071#discussion_r1676961599

Reply:
Push your current package maybe?